interfaces/builtin/serial_port.go

```diff
@@ -43,7 +43,11 @@ func (iface *SerialPortInterface) String() string {
 
 // Pattern to match allowed serial device nodes, path attributes will be
 // compared to this for validity when not using udev identification
-var serialDeviceNodePattern = regexp.MustCompile("^/dev/tty[A-Z]{1,3}[0-9]{1,3}$")
+// Known device node patterns we need to support
+//  - ttyUSBX  (UART over USB devices)
+//  - ttyACMX  (ACM modem devices )
+//  - ttyXRUSBx  (Exar Corp. USB UART devices)
+var serialDeviceNodePattern = regexp.MustCompile("^/dev/tty[a-zA-Z0-9]+[0-9]+$")
```

Comment:
This grants more then we want. Eg:
```
/dev/ttyprintk
/dev/tty0 - ttyN
```

Based on the examples you gave, I suggest this instead:
```
// Known device node patterns we need to support
//  - ttyUSBX  (UART over USB devices)
//  - ttyACMX  (ACM modem devices )
//  - ttyXRUSBX  (Exar Corp. USB UART devices)
//  - ttySX (UART serial ports)
//  - ttyOX (UART serial ports on ARM)
var serialDeviceNodePattern = regexp.MustCompile("^/dev/tty[A-Z]+[0-9]+$")
```
I'd prefer to leave out `[a-z]` unless there is a real-world use case